src/on-create-node.ts

```diff
@@ -37,23 +37,32 @@ const onCreateNode = async (
     throw new Error(`[${response.statusText}]: ${result.message}`)
   }
 
-  const content = JSON.stringify(result.data)
-  const contentDigest = createContentDigest(content)
-  const type = startCase(camelCase(`${node.name}GraphQL`))
-  const child = {
-    ...result.data,
-    id: `__graphql__${contentDigest}`,
-    children: [],
-    parent: node.id,
-    internal: {
-      content,
-      contentDigest,
-      type,
-    },
+  let nodes = [ result.data ]
+  if (map) {
+    nodes = map(result.data)
+    if (!Array.isArray(nodes)) {
+      nodes = [ nodes ]
+    }
   }
 
-  createNode(child)
-  createParentChildLink({ parent: node, child })
+  nodes.forEach(newNode => {
+    const content = JSON.stringify(newNode)
+    const contentDigest = createContentDigest(content)
+    const child = {
+      ...newNode,
+      id: `__graphql__${contentDigest}`,
+      children: [],
+      parent: node.id,
+      internal: {
+        content,
+        contentDigest,
+        type: newNode.type ? newNode.type : startCase(camelCase(`${node.name}GraphQL`)),
```

Comment:
Note that instead of only getting the node type from the file name, we allow the user to define the type as there may be any number of types defined in the results.  We fall back to the file name if type is not provided.